bin/ci

```diff
@@ -103,28 +103,36 @@ function check_tag() {
     site_path="${SITES_DIRECTORY}/${site}"
     tag=$2
 
-    # Tags are of the form "demo-1.2.1" in which "1.2.1" is the version
-    version=${2//${1}-/}
-
     declare -i fail
     fail=0
-    if ! [[ "${tag}" =~ ${site}- ]] ; then
-      echo "Tags must be of the form \"{site}-{semver version}\"!"
-      fail=4
+    if [[ -n "${tag}" ]] ; then
+        # Tags should be of the form "demo-1.2.1" in which "1.2.1" is the version
+        if ! [[ "${tag}" =~ ${site}- ]] ; then
+            echo "Tags must be of the form \"{site}-{semver version}\"!"
+            fail=4
+        fi
+        version=${2//${1}-/}
+    else
+        version=$(grep -m 1 "## \[[0-9]" "${site_path}/CHANGELOG.md" | \
+            cut -d "[" -f2 | cut -d "]" -f1)
     fi
+
     if ! grep "## \[${version}\] - " "${site_path}/CHANGELOG.md" &> /dev/null ; then
       echo "Version \"${version}\" for \"${site}\" is missing from CHANGELOG!"
       fail=4
     fi
+
     init_path="${site_path}/src/backend/${site}/__init__.py"
     if ! grep "__version__ = \"${version}\"" "${init_path}" &> /dev/null ; then
       echo "Backend version does not match the current tag!"
       fail=4
     fi
+
     if ! grep "\"version\": \"${version}\"" "${site_path}/src/frontend/package.json" &> /dev/null ; then
       echo "Frontend version does not match the current tag!"
       fail=4
```

Comment:
Different failures should correspond to different exit codes.

Reply:
I found a way to always be sure the returned code is unique for each different case :shrimp: